Feature/mission 08/워니

ywkim1m opened the pull request:
## 🚀 작업한 기능 설명 (Feature Description)
- 기존 구현 API Swagger 문서 작성

## 🔍 작업 상세 (Implementation Details)
- index.js에서 Swagger doc에 공통 스키마 추가
### 에러 응답
- `schema: { $ref: '#/components/schemas/ErrorResponse' }`
- 모든 API에서 동일한 형태로 문서화, 각 API별로 에러 예시만 example로 바꿈
### 성공 응답
- SuccessResponse로 느슨하게 작성, 각 API에서 example로 실제 데이터 보여줌

## 🖼️ 이미지 첨부 (Images)
<img width="50%" height="50%" alt="image" src="https://github.com/user-attachments/assets/47119149-8ab0-42b8-91b8-380438f65f81" />

## 📋 관련 자료 (Related Resources)
- 8주차 워크북

## 📝 추가 정보 (Additional Information)
- x.

ywkim1m committed:
Fix: mission-07 피드백

- error.js: 에러 클래스 status code 추가
- review.service.js: 커스텀 에러 구현
- userMission.service.js: 미션 유효성 검사 구현

Resolve: #52

ywkim1m committed:
Feat: mission-08 실습

- Swagger doc에 공통 스키마 추가

Resolve: #52

ywkim1m committed:
Feat: mission-08 컨트롤러에서 에러 응답 공통 스키마

- 모든 API에서 동일한 형태로 문서화,
각 API별로 에러 예시만 example로 바꿈
- 성공 응답: 각 API에서 example로 실제 데이터 보여줌

Resolve: #52

KateteDeveloper reviewed (approved):
swagger 정리가 깔끔합니다!! 수고 많으셨습니다:)

soyun0318 reviewed (approved):
스웨거 문서 작성을 잘 하신 것 같아요!

suyeon0421 reviewed (approved):
여러 개의 에러응답까지 구체적으로 잘 작성해주신 것 같아요 수고하셨습니다!!

jeongkyueun reviewed (approved):
스웨거 doc에 공통 스키마 추가하신 거 너무 좋은 거 같습니다 수고 많으셨습니다!

hyeeon reviewed (approved):
실패 응답도 다양하게 잘 작성해주신 것 같아요 수고하셨습니다!

ywkim1m committed:
Docs: Swagger 응답 및 예시 업데이트

- Swagger responses 예시 수정
- $ref 위치 수정
- example 오버라이드 구조 정리
- content -> application/json 형태로 맞춤

Resolve: #52

Hminkyung commented on Chapter_mission/src/controllers/mission.controller.js:
```diff
@@ -1,12 +1,123 @@
+      content: {
+        "application/json": {
+          schema: {
+            $title: "인증샷과 함께 리뷰 남기기",
```
Swagger 문서에서 $는 required property를 의미하는데 이건 현재 OpenAPI 표준과 맞지 않는 형식입니다! 반드시properties, required 로 작성이 되어야 합니당!

Hminkyung replied:
예를 들어서
#swagger.requestBody = {
  required: true,
  content: {
    "application/json": {
      schema: {
        type: "object",
        properties: {
          title: { type: "string" },
          description: { type: "string" },
          point: { type: "integer" },
          deadline: { type: "string", format: "date-time" }
        },
        required: ["title", "description", "point"]
      }
    }
  }
}
>>> 이런식 입니다!

Hminkyung commented on Chapter_mission/src/controllers/review.controller.js:
```diff
@@ -1,13 +1,159 @@
+      description: '리뷰를 등록할 가게의 ID'
+    }
+
+    #swagger.requestBody = {
```
Swagger 주석은 너무 잘 되어있지만 실제 dto 변환시의 검증 로직이 없어요!! 예를 들어서 score은 1~5 사이의 검증이나 required field 등의 검증이 필요할 것 같아요!!

Hminkyung commented on Chapter_mission/src/controllers/user.controller.js:
```diff
@@ -4,6 +4,85 @@ import { userSignUp } from "../services/user.service.js";
+  #swagger.summary = '회원가입'
+  #swagger.description = '이메일, 비밀번호, 닉네임으로 회원을 등록합니다.'
+
+  #swagger.requestBody = {
```
user.service.js 에는 nickname, gender, birth, phoneNumber, profileImage를 prisma에 넣은 상태인데 현재 Swagger에는 없어요! 이런 경우는 오류가 발생할 수 있으므로 Swagger와 실제 필드가 일치해야해요!

Hminkyung reviewed (approved):
미션 수행하느라 너무 고생많으셨어요!! 매주차 피드백도 열심히 반영하시고 항상 열심히 하시는거 너무너무 최고입니당!! 스터디장님 최고 💯

ywkim1m committed:
Feat: Google Login 구현 실습

- Google Login 인증 키 발급 완료
- 성공 응답 확인

Resolve: #62

ywkim1m committed:
Revert "Feat: Google Login 구현 실습"

This reverts commit 8d00ae7677b2d5fb6052bdba98316aa25b732150.
